dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/VisualizationController.java

```diff
@@ -141,6 +150,30 @@ public void postProcessResponseEntity( Visualization visualization, WebOptions o
                     period.setName( i18nFormat.formatPeriod( period ) );
                 }
             }
+
+            List<BaseDimensionalItemObject> dimensionalItemObjectList = visualization.getDataDimensionItems()
+                .stream()
+                .filter( ddi -> ddi.getExpressionDimensionItem() != null )
+                .flatMap( ddi -> ExpressionDimensionItemHelper.getExpressionItems( manager, ddi ).stream() )
+                .collect( Collectors.toList() );
+
+            visualization.getDataDimensionItems().addAll( dimensionalItemObjectList.stream().map( d -> {
+                DataDimensionItem ddi = new DataDimensionItem();
+                if ( d.getDimensionItemType() == DimensionItemType.DATA_ELEMENT )
```

Comment:
Not a big deal, but a switch/case could be better here.

Reply:
done